### PR TITLE
✨ Add AI Team page with hiring wizard and employee agent

### LINKED ISSUE #583: ✨ Standardized message pruning for all chat experiences
## Summary

Implement a unified message pruning layer using Vercel AI SDK v6's `pruneMessages()` that works across all chat contexts: main concierge, AI employees, and background responses.

## Why

Currently, message handling is scattered:
- Main chat filters reasoning via `filterReasoningFromMessages()`
- Workers have separate message prep logic
- No standardized approach to context window management

As we add AI employees (scheduled agents), we need consistent pruning rules everywhere.

### New file: `lib/ai/message-pruning.ts`

```typescript
import { pruneMessages, type ModelMessage } from 'ai';

export interface PruneOptions {
  mode: 'concierge' | 'employee' | 'background';
  maxMessages?: number;
}

export function pruneForAPI(
  messages: ModelMessage[],
  options: PruneOptions
): ModelMessage[] {
  const { mode, maxMessages = 50 } = options;

  // Step 1: Structural pruning (reasoning, old tool calls)
  let pruned = pruneMessages({
    messages,
    reasoning: 'before-last-message',
    toolCalls: 'before-last-2-messages',
    emptyMessages: 'remove',
  });

  // Step 2: Length pruning (keep system + recent)
  if (pruned.length > maxMessages) {
    const system = pruned.filter(m => m.role === 'system');
    const rest = pruned.filter(m => m.role !== 'system');
    pruned = [...system, ...rest.slice(-(maxMessages - system.length))];
  }

  return pruned;
}
```

### Integration points

1. **Main chat** (`app/api/connection/route.ts`) - Use before `streamText()` call
2. **Background worker** (`worker/activities/background-response.ts`) - Use in `generateBackgroundResponse`
3. **AI employees** (new) - Use in `runEmployee()` agent function

## Acceptance Criteria

- [ ] `pruneForAPI()` function exported from `lib/ai/message-pruning.ts`
- [ ] Main chat route uses standardized pruning
- [ ] Background worker uses standardized pruning
- [ ] Existing `filterReasoningFromMessages()` either replaced or composed into new function
- [ ] Tests for pruning edge cases (empty messages, all-system messages, etc.)

## Technical Notes

- Vercel AI SDK v6 provides `pruneMessages()` with options for reasoning, tool calls, and empty message removal
- We're choosing pruning over summarization for v1 (simpler, deterministic)
- Future: Could add LLM-based summarization in `prepareStep` hook if needed

## Related

- Part of AI Team / scheduled agents feature
- Enables shared code between main LLM and AI employees

### PR DESCRIPTION
**Pull request by TechNickAI**
## Summary

Implements the AI Team feature - a "digital chief of staff" experience that transforms Carmenta into a 10x productivity layer through persistent AI agents.

**Key Components:**

- **Message Pruning Layer** - Standardized context management for all chat experiences (closes #583)
- **AI Employee Agent** - Scheduled job execution with full tool access via Vercel AI SDK
- **Command Center** - Activity feed + automation toggles for outcome-focused UX
- **Hiring Wizard** - Conversational interface to describe and "hire" new team members

### Hybrid UX Approach
- **Setup Phase**: Rich conversational wizard (feels like hiring a team member)
- **Ongoing Phase**: Minimal command center (agents become invisible infrastructure)

This follows the spec in `knowledge/components/ai-team.md` - investing in delightful setup while keeping daily interaction effortless.

### Tool Sharing Architecture
The employee agent shares the same tool registry as the main chat (`getIntegrationTools()`). This means scheduled agents have access to all connected integrations automatically.

### Message Pruning Strategy
Using Vercel AI SDK's `pruneMessages()` with mode-specific limits:
- Concierge: 100 messages (full context)
- Employee: 30 messages (focused tasks)
- Background: 50 messages (balanced)

## Test Plan

- [ ] Visit `/ai-team` - should show command center (empty state initially)
- [ ] Click "Hire New" - should open hiring wizard
- [ ] Describe an automation (e.g., "Check my email daily") - wizard should generate playbook
- [ ] Click "Hire This Team Member" - should create job and redirect to command center
- [ ] Toggle automation on/off - should pause/resume schedule
- [ ] When job runs, activity should appear in "What's Happening" feed

## Files Changed

| File | Purpose |
|------|---------|
| `lib/ai/message-pruning.ts` | Standardized message pruning |
| `lib/agents/employee.ts` | AI employee agent with tools |
| `app/ai-team/page.tsx` | Command center dashboard |
| `app/ai-team/hire/page.tsx` | Hiring wizard UI |
| `app/api/ai-team/hire/route.ts` | Wizard API endpoint |
| `app/api/jobs/route.ts` | Added notifications to query |
| `worker/activities/index.ts` | Employee execution activities |

Generated with Carmenta

### CODE REVIEW

File: lib/agents/employee.ts [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+        );
+
+        // Extract completion data from tool call
+        const completeCall = result.steps
+            .flatMap((step) => step.toolCalls ?? [])
+            .find((call) => call.toolName === "complete");
+
+        if (completeCall) {
+            // Access tool call input (the parsed parameters)
+            const completion = (completeCall as { input: CompleteToolParams }).input;
```
[the remark this answers is not in the document]
Comment (TechNickAI) [reply lib/agents/employee.ts]: Fixed in 1e03dfb5 - changed `completeCall.input` to `completeCall.args` with proper type casting through `unknown`. Excellent catch - this would have crashed every employee agent run.

File: worker/workflows/agent-job.ts [hunk 2]
```diff
@@ -26,55 +26,43 @@ export interface AgentJobResult {
...
-        const summary = `Failed: ${errorMessage}`;
 
-        await recordJobRun({
-            jobId,
-            status: "failed",
-            summary,
-            messages,
-        });
+        // Load context again for userId (needed for error recording)
+        const context = await loadFullJobContext(jobId);
```
[the remark this answers is not in the document]
Comment (TechNickAI) [reply worker/workflows/agent-job.ts]: Fixed in 1e03dfb5 - moved context load outside try block and wrapped error recording in try-catch to preserve original error. Prevents cascading failures and infinite retries.
